pkg/cloud/services/ec2/instances.go

```diff
@@ -195,7 +196,17 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*infrav1.Instance,
 		record.Warnf(scope.AWSMachine, corev1.EventTypeWarning, "FailedGetBootstrapData", err.Error())
 		return nil, err
 	}
-	input.UserData = pointer.StringPtr(userData)
+
+	if scope.AWSMachine.Spec.EncryptedCloudInit != nil && *scope.AWSMachine.Spec.EncryptedCloudInit {
```

Comment:
Shouldn't this be better part of the `scope.GetBootstrapData` method?

Reply:
Moved things around a bit, anyway. LMKWYT (let me know what you think) when I push.